ion-schema/src/nfa.rs

```diff
@@ -79,7 +91,9 @@ impl Nfa {
     }
 }
 
-/// A context that will be used for a single transition
+/// A context that will be used to store state-visit counts.
+/// This will be used by [NfaEvaluation] which uses [Nfa] to evaluate given Ion value.
+/// With each element in the given ordered elements there will be a set of [NfaRun]s created for all possible transitions to next states.
```

Comment:
```suggestion
/// For each step in the NFA's evalution, each existing `NfaRun` will be cloned once for each possible transition.
```
Is this accurate? I'm hoping to capture the idea that an NFA run is one of the things moving through the state machine and that it gets cloned a potentially large number of times during evaluation.

Reply:
That is correct. For each possible and correct transition a new `NfaRun` gets created.